File: workers/cartographer/testdata/sample.java
// Sample Java file for Cartographer AST extraction tests.

public class DataProcessor {
    private Config config;

    public DataProcessor(Config config) {
        this.config = config;
    }

    public Object process(Object record) {
        return record;
    }
}

interface Processable {
    Object process(Object record);
}

enum Status {
    ACTIVE,
    INACTIVE
}

class Config {
    private String path;

    public Config(String path) {
        this.path = path;
    }

    public String getPath() {
        return path;
    }
}

File: workers/cartographer/testdata/sample.rs
// Sample Rust file for Cartographer AST extraction tests.

pub struct DataProcessor {
    config: Config,
}

pub struct Config {
    pub path: String,
}

pub enum Status {
    Active,
    Inactive,
}

pub trait Processable {
    fn process(&self, record: &str) -> String;
}

impl DataProcessor {
    pub fn new(config: Config) -> Self {
        DataProcessor { config }
    }
}

impl Processable for DataProcessor {
    fn process(&self, record: &str) -> String {
        record.to_string()
    }
}

pub fn load_config(path: &str) -> Config {
    Config { path: path.to_string() }
}
